src/components/date-picker/date-picker-grid.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import {getDayNames, getDaysInMonth, getNumOfPreviousDays} from './date-picker-helpers';
+import {computed} from 'mobx';
+import {observer} from 'mobx-react';
+import styles from './date-picker.st.css';
+
+export interface DatePickerGridProps {
+    date: Date;
+    startingDay?: number;
+    onChange (selectedDay: string): void;
+}
+
+@observer
+export class DatePickerGrid extends React.Component<DatePickerGridProps, {}> {
+    static defaultProps = {
+        startingDay: 0
+    };
+
+    @computed
+    get dayArray (): Array<JSX.Element> {
+        const dayArray: Array<number> = [];
+        for (let i = 1; i <= getDaysInMonth(this.props.date); i++) {
+            dayArray.push(i);
+        }
+
+        return dayArray.map(day => <span className={[styles.calendarItem, styles.day].join(' ')} onMouseDown={this.onMouseDown} key={'DAY_' + day}
+                                              data-automation-id={'DAY_' + day}>{day}</span>);
```

Comment:
also this find of formating is hard to read. All props should be either in one line or each property in separate line